hie.yaml

```diff
@@ -0,0 +1,106 @@
+      component: "lib:small-steps"
+
+    - path: "semantics/executable-spec/test"
+      component: "small-steps:test:doctests"
+
+    - path: "semantics/small-steps-test/src"
+      component: "lib:small-steps-test"
+
+    - path: "semantics/small-steps-test/test"
+      component: "small-steps-test:test:doctests"
+
+    - path: "semantics/small-steps-test/test"
+      component: "small-steps-test:test:examples"
+
+    - path: "shelley/chain-and-ledger/dependencies/non-integer/src"
+      component: "lib:shelley-spec-non-integral"
+
+    - path: "shelley/chain-and-ledger/dependencies/non-integer/test"
+      component: "shelley-spec-non-integral:test:shelley-spec-non-integral-test"
+
+    - path: "shelley/chain-and-ledger/executable-spec/src"
+      component: "lib:shelley-spec-ledger"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/src"
+      component: "lib:shelley-spec-ledger-test"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test"
+      component: "shelley-spec-ledger-test:test:shelley-spec-ledger-test"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/bench/Main.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test/Main.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/bench/BenchUTxOAggregate.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/bench/BenchValidation.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/bench/Shelley.Spec.Ledger.Bench.Gen.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/bench/Shelley.Spec.Ledger.Bench.Rewards.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/bench/Test.Shelley.Spec.Ledger.Examples.Cast.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test/BenchUTxOAggregate.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test/BenchValidation.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test/Shelley.Spec.Ledger.Bench.Gen.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test/Shelley.Spec.Ledger.Bench.Rewards.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
+
+    - path: "shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test.Shelley.Spec.Ledger.Examples.Cast.hs"
+      component: "shelley-spec-ledger-test:bench:mainbench"
```

Comment:
Does this work as expected? `Test.Shelley.Spec.Ledger.Examples.Cast.hs` doesn't look right, that's a mix of a module name and a file name.

I think it'd be better to not share source files between components.

Reply:
I should probably have actually read my auto-generated file, shouldn't I...

Good catches, I'll fix these up! Although this works for the shelley-spec-ledger package, which is my main goal :-)